components/group-mgt/org.wso2.carbon.device.mgt.group.core/src/main/java/org/wso2/carbon/device/mgt/group/core/dao/GroupDAOImpl.java

```diff
@@ -0,0 +1,199 @@
+        ResultSet rs;
+        int groupId = -1;
+        try {
+            Connection conn = GroupManagementDAOFactory.getConnection();
+            String sql = "INSERT INTO DM_GROUP(DESCRIPTION, NAME, DATE_OF_ENROLLMENT, DATE_OF_LAST_UPDATE, "
+                    + "OWNER, TENANT_ID) VALUES (?, ?, ?, ?, ?, ?)";
+            stmt = conn.prepareStatement(sql);
+            stmt.setString(1, deviceGroup.getDescription());
+            stmt.setString(2, deviceGroup.getName());
+            stmt.setLong(3, new Date().getTime());
+            stmt.setLong(4, new Date().getTime());
+            stmt.setString(5, deviceGroup.getOwner());
+            stmt.setInt(6, tenantId);
+            stmt.executeUpdate();
+            rs = stmt.getGeneratedKeys();
+            if (rs.next()) {
+                groupId = rs.getInt(1);
+            }
+            return groupId;
+        } catch (SQLException e) {
+            String msg = "Error occurred while adding deviceGroup " +
+                    "'" + deviceGroup.getName() + "'";
+            log.error(msg, e);
+            throw new GroupManagementDAOException(msg, e);
+        } finally {
+            GroupManagementDAOUtil.cleanupResources(stmt, null);
+        }
+    }
+
+    @Override public int updateGroup(DeviceGroup deviceGroup) throws GroupManagementDAOException {
+        PreparedStatement stmt = null;
+        int sqlReturn = -1;
+        try {
+            Connection conn = GroupManagementDAOFactory.getConnection();
+            String sql = "UPDATE DM_GROUP SET DESCRIPTION = ?, NAME = ?, DATE_OF_LAST_UPDATE = ?, OWNER = ? "
+                    + "WHERE ID = ?";
+            stmt = conn.prepareStatement(sql);
+            stmt.setString(1, deviceGroup.getDescription());
+            stmt.setString(2, deviceGroup.getName());
+            stmt.setLong(3, deviceGroup.getDateOfLastUpdate());
+            stmt.setString(4, deviceGroup.getOwner());
+            stmt.setInt(5, deviceGroup.getId());
+            sqlReturn = stmt.executeUpdate();
+        } catch (SQLException e) {
+            String msg = "Error occurred while updating deviceGroup " +
+                    "'" + deviceGroup.getName() + "'";
+            log.error(msg, e);
+            throw new GroupManagementDAOException(msg, e);
+        } finally {
+            GroupManagementDAOUtil.cleanupResources(stmt, null);
+        }
+        return sqlReturn;
+    }
+
+    @Override public int deleteGroup(int groupId) throws GroupManagementDAOException {
+        PreparedStatement stmt = null;
+        int sqlReturn = -1;
+        try {
+            Connection conn = GroupManagementDAOFactory.getConnection();
+            String sql = "DELETE FROM DM_GROUP WHERE ID = ?";
+            stmt = conn.prepareStatement(sql);
+            stmt.setInt(1, groupId);
+            sqlReturn = stmt.executeUpdate();
+        } catch (SQLException e) {
+            String msg = "Error occurred while deleting group " +
```

Comment:
Fixed in https://github.com/wso2/carbon-device-mgt/commit/e1cd4b1907f419f974545fc214c58cd9f247f2b2